keyserver/handler.go

```diff
@@ -28,18 +28,28 @@ func (server *ConiksServer) handleRegistrationMessage(reg *p.RegistrationRequest
 			p.Error(p.ErrorNameExisted)
 	}
 
-	tb, errCode := server.directory.Register(reg.Username, key)
+	ap, tb, errCode := server.directory.Register(reg.Username, key)
 	if errCode != p.Success {
 		server.Unlock()
 		return p.NewErrorResponse(errCode), p.Error(errCode)
 	}
 	server.tbs[reg.Username] = tb
 	server.Unlock()
 
+	// TODO: do we want to store the user policies into DB?
+
 	tbEncoded, err := p.MarshalTemporaryBinding(tb)
 	if err != nil {
 		return p.NewErrorResponse(p.ErrorInternalServer), err
 	}
+	apEncoded, err := p.MarshalAuthenticationPath(ap)
+	if err != nil {
+		return p.NewErrorResponse(p.ErrorInternalServer), err
+	}
 
-	return &p.RegistrationResponse{TempraryBinding: string(tbEncoded)}, nil
+	return &p.RegistrationResponse{
+		RegEpoch: server.directory.LatestEpoch(),
```

Comment:
Shouldn't this be the next epoch since the registration won't appear in the tree until then?